pkg/service/group_service_factory.go

```diff
@@ -72,6 +74,26 @@ func NewNsGroupService(ip, username, password, apiVersion string, synchronous bo
 	return &NsGroupService{ip: ip, client: client}, nil
 }
 
+// DeleteNsGroupService - delete the client object and remove the session token
+func (gs *NsGroupService) DeleteNsGroupService() error {
+	tokenService := gs.GetTokenService()
+	sessionToken := gs.client.SessionToken
+
+	// apply filter on session token
+	sFilter := &param.GetParams{
+		Filter: &param.SearchFilter{
+			FieldName: nimbleos.TokenFields.SessionToken,
+			Operator:  param.EQUALS.String(),
+			Value:     sessionToken,
+		},
+	}
+	tokenList, err := tokenService.GetTokens(sFilter)
+	if err != nil {
+		return err
+	}
+	return tokenService.DeleteToken(*tokenList[0].ID)
+}
+
```

Comment:
This doesn't seem correct to me.  I prefer you just delete the token through the tokenService.  Alternatively, you can rename this function to logout I suppose.

Reply:
Ok will change to LogoutService